closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -612,9 +618,37 @@ private static void restrictsCheck() {
             printStackTraceAndExit("Restricted security property is null.");
         }
 
-        // Check if the SunsetDate expired.
-        if (isPolicySunset(restricts.descSunsetDate)) {
-            printStackTraceAndExit("Restricted security policy expired.");
+        int expireMonths = monthsToPolicySunset(restricts.descSunsetDate);
+
+        if (suppressSunsetWarning) {
+            if (expireMonths == 0 && !ignoreSunsetExpiration) {
+                System.exit(1);
+            }
+        } else {
+            if (expireMonths == 0) { // Check if the SunsetDate expired.
+                if (ignoreSunsetExpiration) {
+                    System.err.println("The requested restricted security profile " + restricts.profileID
+                            + " expired on " + restricts.descSunsetDate
+                            + ": java will stop because certified cryptography use cannot be guaranteed."
+                            + " Use -Dsemeru.restrictedsecurity.suppresssunsetwarning to stop displaying this message."
+                            + " The -Dsemeru.restrictedsecurity.ignoresunsetexpiration option has been specified."
+                            + " WARNING: java will start with the requested restricted security profile but uncertified"
+                            + " cryptography may be active.");
```

Comment:
This message is not self-consistent. It says "java will stop" and "java will start" - it can't be both.

Reply:
Updated.